LiteFarmOrg/LiteFarm
# Fix/task reassign and assign all on date bug

Ticket: [LF-2323](https://lite-farm.atlassian.net/browse/LF-2323) (see most recent comment)

This pull request was made to allow users to reassign a task to a user and assign all unassigned tasks to the same user when "assign all unassigned tasks on this date to this person" is checked. 

**Summary of Changes:** 
- Refactored assignTask endpoint so that logic can be shared with assignAllTasksOnDate endpoint
- Made it so that reassigning a task is possible from assignAllTasksOnDate endpoint
- Added tests for this case


**To Test:**
- Run backend tests in task.test.js, make sure they pass
- Have at least two users at a farm
- Create a few tasks (maybe about 4) with at least two being on the same date
- Assign one of the tasks on the same date to a user
- Make sure that another task on the same date is unassigned 
- Reassign that task to a different user while having "assign all unassigned tasks on this date to this person" checked
- Ensure that the previously unassigned tasks on the same date and the task you reassigned are all assigned to the intended user
- Play around with assigning, unassigning, and assigning all tasks on date to make sure there are no other related bugs
